day1/part2.py

```diff
@@ -0,0 +1,43 @@
+with open('input.txt', 'r') as f:
+    numbers = ('one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine')
+    lines = list(map(str.rstrip, f.readlines()))
+    num_of_line = 0
+    counter = 0
+    for line in lines:
+        counter += 1
+        for character in line:
+            if character.isdigit():
+                first = {character: line.index(character)}
+                break
+        for number in numbers:
+            if line.find(number) != -1:
+                first.update({number: line.index(number)})
+
+        for character in line[::-1]:
+            if character.isdigit():
+                last = {character: line[::-1].index(character)}
+                break
+        for number in numbers:
+            if line.find(number) != -1:
+                last.update({number: line[::-1].index(number[::-1])})
+
+
+        tmp = min(first, key=first.get)
```

Comment:
Same code twice + this also seems kind of dubious rather create a separate function call it twice and add the results. 🥶